### PR TITLE
Update docs.

### PR DESCRIPTION
**Pull request by olafurpg**
- Port DisableSyntax and MissingFinal to the new doc format.
- Make MissingFinal configurable so that it can be documented like other rules.
- Automatically render rules.md page with all rule names + description
- Deprecate NoFinalize in favor of DisableSyntax.noFinalize. I did this here because I noticed `NoFinalize` was missing a description which made rules.md ugly. 
- Best-effort pretty-printing of default values.

New rules overview

screen shot 2018-01-26 at 12 01 16

### CODE REVIEW

**Comment by olafurpg** [comment]
cc/ @vovapolu DisableSyntax page has grown a lot!
screen shot 2018-01-26 at 12 01 29

**Comment by olafurpg** [comment]
@vovapolu I made MissingFinal configurable, 
screen shot 2018-01-26 at 12 05 14

**Comment by olafurpg** [comment]
The new overview page is inspired by https://eslint.org/docs/rules/

**Comment by olafurpg** [comment]
@ag91 FYI, your #422 PR is now being used to render the rules.md page!

**Comment by MasseGuillaume** [comment]
@olafurpg That was productive, we should update docs more often!

File: scalafix-core/shared/src/main/scala/scalafix/internal/rule/NoFinalize.scala [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
 
 import scala.meta._
-
 import scalafix.rule.Rule
 import scalafix.rule.RuleCtx
 import scalafix.lint.LintMessage
-import scalafix.lint.LintCategory
+import scalafix.rule.RuleName
+
+case object NoFinalize
```
Comment (MasseGuillaume) [inline scalafix-core/shared/src/main/scala/scalafix/internal/rule/NoFinalize.scala]: @olafurpg We can delete this one, it was never released.

File: website/src/main/scala/scalafix/website/package.scala [hunk 2]
```diff
@@ -1,11 +1,58 @@
...
+          th("Name"),
+          th("Description")
+        )
+      ),
+      tbody(rows)
+    )
+    html.toString()
+  }
+
+  // TODO(olafur) replace this hack with ConfEncoder[T] typeclass.
```
Comment (MasseGuillaume) [inline website/src/main/scala/scalafix/website/package.scala]: I'm going to add a scalafix Rule to disallow TODO in comments ;-)

**Comment by MasseGuillaume** [comment]
Nevermind `NoFinalize` was added in 0.5.7 https://github.com/scalacenter/scalafix/blob/v0.5.7/scalafix-core/shared/src/main/scala/scalafix/internal/rule/NoFinalize.scala

File: website/src/main/scala/scalafix/website/package.scala (hunk 2 above)
Comment (MasseGuillaume) [reply website/src/main/scala/scalafix/website/package.scala]: I guess we can do this with the `DisableSyntax.regex`

**Comment by vovapolu** [comment]
Very cool!

**Pull request merged**